### DISCUSSION TITLE
Add a short delay before expanding sidebar.

### DISCUSSION THREAD

**Discussion by StDensity** [zen-browser/desktop#3358] [Ideas]
Currently if you are on compact mode and turned on the hover to expand side bar then it's kinda messy. Because when you try to click the icon, it suddenly expands and changes the position.  
Eg:  
<video src="https://github.com/user-attachments/assets/30796d49-369e-4434-9377-b87ea6a72b30" controls></video>

MS Edge has a delay before expanding and it's really convenient when you just want to click the icon without expanding the sidebar.  
Eg:  
<video src="https://github.com/user-attachments/assets/5eda20bd-370e-4af1-9059-7da4c3e85755" controls></video>

It would be cool to have a mod or a feature implementing this.

**Comment by dnut** [comment]
I agree there's a problem. In your video, that does not look right. If the items on the bottom are already stacked vertically, they should probably stretch horizontally instead of rearranging.

But I am strongly opposed to imposing a delay on the sidebar opening. If anything, it should open faster. Waiting for the animation is already annoying.

Adding an option for a delay would be fine. As long as 0 delay is always allowed.
